Samples/Emoji8/UWP/cs/Emoji8/Services/IntelligenceService.cs

```diff
@@ -0,0 +1,297 @@
+﻿// Copyright (c) Microsoft Corporation. 
+// Licensed under the MIT license. 
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using Windows.AI.MachineLearning;
+using Windows.Graphics.Imaging;
+using Windows.Media;
+using Windows.Storage;
+using Windows.Media.FaceAnalysis;
+using Windows.UI.Xaml.Media.Imaging;
+using Windows.UI.Core;
+
+namespace Emoji8.Services
+{
+    public class IntelligenceService
+    {
+        public event EventHandler IntelligenceServiceProcessingCompleted;
+        public event EventHandler<ClassifiedEmojiEventArgs> IntelligenceServiceEmotionClassified;
+
+        private bool _isInitialized = false;
+        readonly DateTime _lastPositiveIdentification = DateTime.Now;
+        private FaceDetector _faceDetector;
+
+        private IntelligenceService() { }
+
+        private static IntelligenceService _current;
+
+        public static IntelligenceService Current => _current ?? (_current = new IntelligenceService());
+
+        private LearningModel _model = null;
+        private LearningModelDevice _device;
+        private LearningModelSession _session;
+        private TensorFeatureDescriptor _inputImageDescriptor;
+        private TensorFeatureDescriptor _outputTensorDescriptor;
+        private LearningModelBinding _binding;
+
+        public async Task<bool> InitializeAsync()
+        {
+            bool modelLoaded = await LoadModelAsync();
+            if (modelLoaded == true)
+            {
+                _faceDetector = await FaceDetector.CreateAsync();
+                CameraService.Current.SoftwareBitmapFrameCaptured += Current_SoftwareBitmapFrameCaptured;
+                _isInitialized = true;
+            }
+            return modelLoaded;
+        }
+
+        private async Task<bool> LoadModelAsync()
+        {
+            var modelStorageFile = await StorageFile.GetFileFromApplicationUriAsync(new Uri(Constants.MODEL_PATH));
+
+            try
+            {
+                _model = await LearningModel.LoadFromStorageFileAsync(modelStorageFile);
+            }
+            catch(Exception ex)
+            {
+                Debug.WriteLine(ex.Message);
+            }
+
+            _device = new LearningModelDevice(LearningModelDeviceKind.Default);
```

Comment:
if you're just going to use the default device, you actually don't need to specify it - you can just create a session without the device parameter: "_session = new LearningModelSession(_model);".
alternatively, you could try to run this on the GPU too if the user has one.